### PR TITLE
Feature/coach season

### PR DESCRIPTION
**Pull request by garrettgregor**
Further refactor is certainly possible, but I just don't have enough time today to go back through and create additional helper methods and tests and think it is best if we move forward as is and focus on the presentation topics for tomorrow.

✅ #winningest_coach and #worst_coach working and all tests passing
✅ spec_harness pass minus methods not yet written from iteration 4
✅ RSpec: 100%: `466 / 466 LOC (100.0%) covered.`
🤔 Rubocop
  - update to previous test to conform to `_` in place of `,` in test for large numbers
  -  errors below; don't have time to refactor to conform:

```bash
Offenses:

lib/stat_tracker.rb:6:1: C: Metrics/ClassLength: Class has too many lines. [260/100]
class StatTracker ...
^^^^^^^^^^^^^^^^^
lib/stat_tracker.rb:65:31: C: [Correctable] Style/SymbolProc: Pass &:season as an argument to map instead of a block.
    season_count = @games.map { |game| game.season }
                              ^^^^^^^^^^^^^^^^^^^^^^
lib/stat_tracker.rb:77:3: C: Metrics/AbcSize: Assignment Branch Condition size for average_goals_by_season is too high. [<7, 16, 2> 17.58/17]
  def average_goals_by_season ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/stat_tracker.rb:155:3: C: Metrics/AbcSize: Assignment Branch Condition size for winningest_coach is too high. [<9, 17, 4> 19.65/17]
  def winningest_coach(season) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/stat_tracker.rb:155:3: C: Metrics/MethodLength: Method has too many lines. [13/10]
  def winningest_coach(season) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/stat_tracker.rb:163:7: C: [Correctable] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
      if game.result == "WIN"
      ^^
lib/stat_tracker.rb:175:3: C: Metrics/AbcSize: Assignment Branch Condition size for worst_coach is too high. [<10, 19, 5> 22.05/17]
  def worst_coach(season) ...
  ^^^^^^^^^^^^^^^^^^^^^^^
lib/stat_tracker.rb:175:3: C: Metrics/MethodLength: Method has too many lines. [15/10]
  def worst_coach(season) ...
  ^^^^^^^^^^^^^^^^^^^^^^^
lib/stat_tracker.rb:184:7: C: [Correctable] Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
      if game.result == "WIN" ...
      ^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 9 offenses detected, 3 offenses autocorrectable
```

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -152,6 +152,49 @@ def lowest_scoring_home_team
...
+        coach_wins_for_season[game.head_coach] += 1
+      end
+    end
+
+    winningest_coach = coach_wins_for_season.max_by do |coach, wins|
+      (wins.to_f / games_coached_for_season[coach]).round(2)
+    end
+
+    winningest_coach[0]
+  end
```
Comment (javenb022) [inline lib/stat_tracker.rb:171]: Could be refactored, but with time constraints, this works and looks good

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -152,6 +152,49 @@ def lowest_scoring_home_team
...
+        coach_wins_for_season[game.head_coach] += 0
+      end
+    end
+
+    worst_coach = coach_wins_for_season.min_by do |coach, wins|
+      (wins.to_f / games_coached_for_season[coach]).round(2)
+    end
+
+    worst_coach[0]
+  end
```
Comment (javenb022) [inline lib/stat_tracker.rb:189]: Same thing with previous comment

**Review by javenb022 — APPROVED** [review]
Looking delicious

**Review by MylesNottingham — APPROVED** [review]
We worked through this together as a team over slack huddle and it works great.

**Pull request merged**